--- curljava.c.orig	2008-10-20 09:34:12 +0000
+++ curljava.c	2013-02-11 17:10:00 +0000
@@ -116,7 +116,7 @@
   void *handle = (void *)((struct curljava*)jcurl)->libcurl;
 
 #ifdef DEBUG
-  puts("setopt int + string");
+  fputs("setopt int + string", stderr);
 #endif
 
   return (jint)curl_easy_setopt(handle, (CURLoption)option, str);
@@ -133,7 +133,7 @@
   CURLoption opt = (CURLoption)option;
 
 #ifdef DEBUG
-  puts("setopt int + int");
+  fputs("setopt int + int", stderr);
 #endif
 
   switch(opt) {
@@ -148,7 +148,7 @@
   return (jint) curl_easy_setopt(handle, (CURLoption) option, value);
 }
 
-static int curljava_write_callback(void *ptr,
+static size_t curljava_write_callback(void *ptr,
                                    size_t size,
                                    size_t nmemb,
                                    FILE  *stream)
@@ -156,8 +156,8 @@
   struct curljava *curl = (struct curljava *)stream;
   size_t realsize = size * nmemb;
   JNIEnv *java = curl->write.java;
-  jbyteArray jb=NULL;
-  int ret=0;
+  jbyteArray jb = NULL;
+  int ret = 0;
 
 #ifdef DEBUG
   fprintf(stderr, "%d bytes data received in callback:\n"
@@ -180,6 +180,9 @@
 
 #ifdef DEBUG
   fprintf(stderr, "java-method returned %d\n", ret);
+#else
+  /* kill warning */
+  (void)ret;
 #endif
 
   return realsize;
@@ -203,16 +206,16 @@
     /* this makes a reference that'll be alive until we kill it! */
     cls = (*java)->NewGlobalRef(java, cls_local);
 
-#ifdef DEBUG
-    printf("setopt int + object, option = %d cls= %p\n",
-           option, cls);
-#endif
-
     if(!cls) {
       fputs("couldn't make local reference global\n", stderr);
       return 0;
     }
 
+#ifdef DEBUG
+    fprintf(stderr, "setopt int + object, option = %d cls = %p\n",
+            option, cls);
+#endif
+
     /* this is the write callback */
     mid = (*java)->GetMethodID(java, cls, "handleString", "([B)I");
     if(!mid) {
@@ -244,7 +247,7 @@
 }
 
 
-static int curljava_read_callback(void *ptr,
+static size_t curljava_read_callback(void *ptr,
                                   size_t size,
                                   size_t nmemb,
                                   FILE  *stream)
@@ -286,16 +289,16 @@
     /* this makes a reference that'll be alive until we kill it! */
     cls = (*java)->NewGlobalRef(java, cls_local);
 
-#ifdef DEBUG
-    printf("setopt int + object, option = %d cls= %p\n",
-           option, cls);
-#endif
-
     if(!cls) {
       fputs("couldn't make local reference global\n", stderr);
       return 0;
     }
 
+#ifdef DEBUG
+    fprintf(stderr, "setopt int + object, option = %d cls = %p\n",
+            option, cls);
+#endif
+
     /* this is the read callback */
     mid = (*java)->GetMethodID(java, cls, "retrieveString", "([B)I");
     if(!mid) {
