--- MakeCurlGlue.java.orig	2008-10-20 09:34:12 +0000
+++ MakeCurlGlue.java	2013-02-11 16:07:39 +0000
@@ -77,8 +77,16 @@
     }
     fis.close();
 
-    s= "  // end of generated list" + ls + ls;
+    s = "  // end of generated list" + ls + ls;
     System.out.print( s );
+
+    s = "  // CURL_SSLVERSION constants" + ls +
+        "  public static final int CURL_SSLVERSION_DEFAULT = 0;" + ls +
+        "  public static final int CURL_SSLVERSION_TLSv1 = 1;" + ls +
+        "  public static final int CURL_SSLVERSION_SSLv2 = 2;" + ls +
+        "  public static final int CURL_SSLVERSION_SSLv3 = 3;" + ls + ls;
+    System.out.print( s );
+
     s = "  public CurlGlue() {" + ls +
         "    try {" + ls +
         "      curljava_handle = jni_init();" + ls +
