--- MakeCurlGlue.pl.orig	2008-10-20 09:34:12 +0000
+++ MakeCurlGlue.pl	2013-02-11 17:23:31 +0000
@@ -53,6 +53,12 @@
 print <<EOTXT;
   // end of generated list
 
+  // CURL_SSLVERSION constants
+  public static final int CURL_SSLVERSION_DEFAULT = 0;
+  public static final int CURL_SSLVERSION_TLSv1 = 1;
+  public static final int CURL_SSLVERSION_SSLv2 = 2;
+  public static final int CURL_SSLVERSION_SSLv3 = 3;
+
   public CurlGlue() {
     try {
       curljava_handle = jni_init();
