--- test.java.orig	2008-10-20 09:34:12 +0000
+++ test.java	2013-02-11 16:39:53 +0000
@@ -35,12 +35,25 @@
         String pinCode = new String("1234");
 
         boolean bInsecureMode = false;
-        if( args.length == 1 )
-            bInsecureMode = ( args[0].compareTo("-k") == 0 ) ||
-                            ( args[0].compareTo("--insecure") == 0 );
+        boolean bVerboseMode = false;
+        int iSSLVersion = CurlGlue.CURL_SSLVERSION_DEFAULT;
 
+        for (int i = 0; i < args.length; i++) {
+            if (( args[i].compareTo("-k") == 0 ) || ( args[i].compareTo("--insecure") == 0 )) {
+                bInsecureMode = true;
+            } else if (( args[i].compareTo("-v") == 0 ) || ( args[i].compareTo("--verbose") == 0 )) {
+                bVerboseMode = true;
+            } else if (( args[i].compareTo("-1") == 0 ) || ( args[i].compareTo("--tlsv1") == 0 )) {
+                iSSLVersion = CurlGlue.CURL_SSLVERSION_TLSv1;
+            } else if (( args[i].compareTo("-2") == 0 ) || ( args[i].compareTo("--sslv2") == 0 )) {
+                iSSLVersion = CurlGlue.CURL_SSLVERSION_SSLv2;
+            } else if (( args[i].compareTo("-3") == 0 ) || ( args[i].compareTo("--sslv3") == 0 )) {
+                iSSLVersion = CurlGlue.CURL_SSLVERSION_SSLv3;
+            }
+        }
         // System.out.println( "libcurl version: " + CurlGlue.version() );
         System.out.println( "Using " + (bInsecureMode ? "in" : "") + "secure certificate mode." );
+        System.out.println( "Using " + (bVerboseMode ? "" : "non-") + "verbose mode." );
 
         CurlGlue cg;
 
@@ -53,9 +66,9 @@
 
             // Login to the bank's secure Web site, posting account number and PIN code
             cg.setopt(CurlGlue.CURLOPT_URL, "https://www.santander.com.mx/SuperNetII/servlet/Login");
-            cg.setopt(CurlGlue.CURLOPT_SSLVERSION, 3);
+            cg.setopt(CurlGlue.CURLOPT_SSLVERSION, iSSLVersion);
             cg.setopt(CurlGlue.CURLOPT_SSL_VERIFYPEER, bInsecureMode ? 0 : 1);
-            cg.setopt(CurlGlue.CURLOPT_VERBOSE, 1);
+            cg.setopt(CurlGlue.CURLOPT_VERBOSE, bVerboseMode ? 1 : 0);
             cg.setopt(CurlGlue.CURLOPT_FOLLOWLOCATION, 1);
             cg.setopt(CurlGlue.CURLOPT_POST, 1);
             cg.setopt(CurlGlue.CURLOPT_COOKIEJAR, "cookie.txt");
@@ -64,9 +77,9 @@
 
             // Access the bank account balance re-using the session ID stored in memory inside a cookie
             cg.setopt(CurlGlue.CURLOPT_URL, "https://www.santander.com.mx/SuperNetII/servlet/Cchequeras");
-            cg.setopt(CurlGlue.CURLOPT_SSLVERSION, 3);
+            cg.setopt(CurlGlue.CURLOPT_SSLVERSION, iSSLVersion);
             cg.setopt(CurlGlue.CURLOPT_SSL_VERIFYPEER, bInsecureMode ? 0 : 1);
-            cg.setopt(CurlGlue.CURLOPT_VERBOSE, 1);
+            cg.setopt(CurlGlue.CURLOPT_VERBOSE, bVerboseMode ? 1 : 0);
             cg.setopt(CurlGlue.CURLOPT_FOLLOWLOCATION, 1);
             cg.setopt(CurlGlue.CURLOPT_COOKIEFILE, "cookie.txt");
             cg.perform();
